riscv/cfg.h

```diff
@@ -25,24 +27,46 @@ class cfg_arg_t {
   bool was_set;
 };
 
+// Configuration that describes a memory region
+class mem_cfg_t
+{
+public:
+  mem_cfg_t(reg_t base, reg_t size)
+    : base(base), size(size)
+  {
+    // The truth of these assertions should be ensured by whatever is creating
+    // the regions in the first place, but we have them here to make sure that
+    // we can't end up describing memory regions that don't make sense. They
+    // ask that size is a multiple of the minimum page size, that it is
+    // positive and that the top address is still representable in a reg_t.
+    assert((size % PGSIZE == 0) && (base + size > base));
```

Comment:
Can we also add a `(base % PGSIZE == 0)` here?

Reply:
Good point! Done.